src/sql/parts/jobManagement/views/jobsView.component.ts

```diff
@@ -172,50 +192,135 @@ export class JobsViewComponent implements AfterContentChecked {
 	}
 
 	private onJobsAvailable(jobs: sqlops.AgentJobInfo[]) {
-		let jobViews = jobs.map((job) => {
-			return {
-				id: job.jobId,
-				jobId: job.jobId,
-				name: job.name,
-				lastRun: AgentJobUtilities.convertToLastRun(job.lastRun),
-				nextRun: AgentJobUtilities.convertToNextRun(job.nextRun),
-				enabled: AgentJobUtilities.convertToResponse(job.enabled),
-				currentExecutionStatus: AgentJobUtilities.convertToExecutionStatusString(job.currentExecutionStatus),
-				category: job.category,
-				runnable: AgentJobUtilities.convertToResponse(job.runnable),
-				hasSchedule: AgentJobUtilities.convertToResponse(job.hasSchedule),
-				lastRunOutcome: AgentJobUtilities.convertToStatusString(job.lastRunOutcome)
-			};
-		});
+		let jobViews: any;
+		if (!jobs) {
+			let dataView = this._jobCacheObject.dataView;
+			jobViews = dataView.getItems();
+		} else {
+			jobViews = jobs.map((job) => {
+				return {
+					id: job.jobId,
+					jobId: job.jobId,
+					name: job.name,
+					lastRun: AgentJobUtilities.convertToLastRun(job.lastRun),
+					nextRun: AgentJobUtilities.convertToNextRun(job.nextRun),
+					enabled: AgentJobUtilities.convertToResponse(job.enabled),
+					currentExecutionStatus: AgentJobUtilities.convertToExecutionStatusString(job.currentExecutionStatus),
+					category: job.category,
+					runnable: AgentJobUtilities.convertToResponse(job.runnable),
+					hasSchedule: AgentJobUtilities.convertToResponse(job.hasSchedule),
+					lastRunOutcome: AgentJobUtilities.convertToStatusString(job.lastRunOutcome)
+				};
+			});
+		}
 		this._table.registerPlugin(<any>this.rowDetail);
+		this.filterPlugin.onFilterApplied.subscribe((e, args) => {
```

Comment:
It might make more sense to refactor this code out as a class and break it up into different function/parts to make it more readble